charts/gateway-helm/templates/envoy-gateway-deployment.yaml

```diff
@@ -69,7 +69,7 @@ spec:
         - --v=0
         env:
         - name: KUBERNETES_CLUSTER_DOMAIN
-          value: {{ .Values.kubernetesClusterDomain }}
```

Comment:
why remove this?

Reply:
Imo we need to treat helm values as APi fields and expose them when needed, else we cause breaking changes, the goal of this PR is to expose minimum helm values needed for 0.4.0

Reply:
but this need be configurable in the future, xref: https://github.com/envoyproxy/gateway/issues/1280